File: app/src/main/java/com/soya/launcher/ui/dialog/AppDialog.java
package com.soya.launcher.ui.dialog;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.content.pm.ApplicationInfo;
import android.content.pm.PackageInstaller;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.util.Log;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.soya.launcher.R;
import com.soya.launcher.enums.Atts;
import com.soya.launcher.enums.IntentAction;
import com.soya.launcher.utils.AndroidSystem;

public class AppDialog extends SingleDialogFragment implements View.OnClickListener {
    public static final String TAG = "AppDialog";
    public static AppDialog newInstance(ApplicationInfo info) {

        Bundle args = new Bundle();
        args.putParcelable(Atts.BEAN, info);
        AppDialog fragment = new AppDialog();
        fragment.setArguments(args);
        return fragment;
    }

    private View mCloseView;
    private View mDeleteView;
    private View mOpenView;

    private ImageView mIV;
    private TextView mNameView;
    private TextView mVersionView;
    private ImageView mBlur;
    private View mRootView;

    private ApplicationInfo info;

    private Callback callback;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        info = getArguments().getParcelable(Atts.BEAN);
    }

    @Override
    protected int getLayout() {
        return R.layout.dialog_app;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        mOpenView.requestFocus();
        //blur(mRootView, mBlur);
    }

    @Override
    protected void init(LayoutInflater inflater, View view) {
        super.init(inflater, view);
        mCloseView = view.findViewById(R.id.close);
        mDeleteView = view.findViewById(R.id.delete);
        mOpenView = view.findViewById(R.id.open);
        mIV = view.findViewById(R.id.icon);
        mNameView = view.findViewById(R.id.name);
        mVersionView = view.findViewById(R.id.version);
        mBlur = view.findViewById(R.id.blur);
        mRootView = view.findViewById(R.id.root);
    }

    @Override
    protected void initBefore(LayoutInflater inflater, View view) {
        super.initBefore(inflater, view);
        mDeleteView.setVisibility(AndroidSystem.isSystemApp(info.flags) ? View.GONE : View.VISIBLE);
        mCloseView.setOnClickListener(this);
        mDeleteView.setOnClickListener(this);
        mOpenView.setOnClickListener(this);
    }

    @Override
    protected void initBind(LayoutInflater inflater, View view) {
        super.initBind(inflater, view);
        try {
            PackageManager pm = getActivity().getPackageManager();
            mIV.setImageDrawable(info.loadIcon(pm));
            mNameView.setText(info.loadLabel(pm));
            mVersionView.setText(pm.getPackageInfo(info.packageName, 0).versionName);
        } catch (PackageManager.NameNotFoundException e) {
            throw new RuntimeException(e);
        }
    }

    @Override
    protected int[] getWidthAndHeight() {
        return new int[]{ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT};
    }

    @Override
    protected float getDimAmount() {
        return 0;
    }

    @Override
    protected int getGravity() {
        return Gravity.CENTER;
    }

    @Override
    public boolean isMaterial() {
        return false;
    }

    @Override
    public void onClick(View v) {
        if (v.equals(mCloseView)){
            dismiss();
        }else if (v.equals(mDeleteView)){
            //UninstallDialog.newInstance(info).show(getActivity().getSupportFragmentManager(), UninstallDialog.TAG);

            AndroidSystem.uninstallPackage(getActivity(), info.packageName);
            dismiss();
        }else if (v.equals(mOpenView)){
            if (callback != null) callback.onOpen();
            dismiss();
        }
    }

    public void setCallback(Callback callback) {
        this.callback = callback;
    }

    public interface Callback{
        void onOpen();
    }
}

File: app/src/main/java/com/soya/launcher/ui/dialog/AppDialog.kt
package com.soya.launcher.ui.dialog

import android.content.pm.ApplicationInfo
import android.content.pm.PackageManager
import android.os.Bundle
import android.view.Gravity
import android.view.View
import android.view.ViewGroup
import com.shudong.lib_base.ext.e
import com.shudong.lib_base.ext.yes
import com.soya.launcher.databinding.DialogAppBinding
import com.soya.launcher.enums.Atts
import com.soya.launcher.ext.uninstallApkForNormalApp
import com.soya.launcher.ext.uninstallApp
import com.soya.launcher.utils.AndroidSystem.isSystemApp

class AppDialog : SingleDialogFragment<DialogAppBinding>(), View.OnClickListener {


    private var info: ApplicationInfo? = null

    private var callback: Callback? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        info = requireArguments().getParcelable(Atts.BEAN)
    }



    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.open.requestFocus()
    }

    override fun init(view: View) {

        blur(binding.root, binding.blur)
    }

    override fun initBefore(view: View) {
        binding.delete.visibility =
            if (isSystemApp(info!!.flags)) View.GONE else View.VISIBLE
        binding.close.setOnClickListener(this)
        binding.delete.setOnClickListener(this)
        binding.open.setOnClickListener(this)
    }

    override fun initBind( view: View) {
        try {
            val pm = requireActivity().packageManager
            binding.icon.setImageDrawable(info!!.loadIcon(pm))
            binding.name.text = info!!.loadLabel(pm)
            binding.version.text = pm.getPackageInfo(info!!.packageName, 0).versionName
        } catch (e: PackageManager.NameNotFoundException) {
            throw RuntimeException(e)
        }
    }

    override fun getWidthAndHeight(): Pair<Int,Int> {
        return ViewGroup.LayoutParams.MATCH_PARENT to ViewGroup.LayoutParams.MATCH_PARENT
    }

    override fun getDimAmount(): Float {
        return 0f
    }

    override fun getGravity(): Int {
        return Gravity.CENTER
    }

    override fun isMaterial(): Boolean {
        return false
    }

    override fun onClick(v: View) {
        if (v == binding.close) {
            dismiss()
        } else if (v == binding.delete) {
            //UninstallDialog.newInstance(info).show(getActivity().getSupportFragmentManager(), UninstallDialog.TAG);

           // uninstallPackage(requireActivity(), info!!.packageName)
            /*info?.packageName?.let {
                it.uninstallApp{isUninstallSuccess->
                    isUninstallSuccess.yes {
                    }
                }
            }*/
            "开始进行卸载操作：${info?.packageName}".e("chihi_error")
            info?.packageName?.uninstallApkForNormalApp()

            dismiss()
        } else if (v == binding.open) {
            if (callback != null) callback!!.onOpen()
            dismiss()
        }
    }

    fun setCallback(callback: Callback?) {
        this.callback = callback
    }

    interface Callback {
        fun onOpen()
    }

    companion object {
        const val TAG: String = "AppDialog"
        fun newInstance(info: ApplicationInfo?): AppDialog {
            val args = Bundle()
            args.putParcelable(Atts.BEAN, info)
            val fragment = AppDialog()
            fragment.arguments = args
            return fragment
        }
    }
}
